egs/chime4/s5_1ch/local/chime4_train_lms.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+
+# Modified from the script for CHiME3 baseline
+# Copyright 2015, Mitsubishi Electric Research Laboratories, MERL (Author: Takaaki Hori)
+
+# Config:
+order=5 # n-gram order
+
+. utils/parse_options.sh || exit 1;
+
+. ./path.sh
+
+if [ $# -ne 1 ]; then
+  printf "\nUSAGE: %s <Chime4 root directory>\n\n" `basename $0`
+  echo "Please specifies a Chime4 root directory"
+  echo "If you use kaldi scripts distributed in the Chime4 data,"
+  echo "It would be `pwd`/../.."
+  exit 1;
+fi
+
+# check data directories
+chime4_data=$1
+wsj0_data=$chime4_data/data/WSJ0 # directory of WSJ0 in Chime4. You can also specify your WSJ0 corpus directory
+if [ ! -d $chime4_data ]; then
+  echo "$chime4_data does not exist. Please specify chime4 data root correctly" && exit 1
+fi
+if [ ! -d $wsj0_data ]; then
+  echo "$wsj0_data does not exist. Please specify WSJ0 corpus directory" && exit 1
+fi
+lm_train=$wsj0_data/wsj0/doc/lng_modl/lm_train/np_data
+
+# check whether run_init is executed
+if [ ! -d data/lang ]; then
+  echo "error, execute local/run_init.sh, first"
+  exit 1;
+fi
+
+# lm directories
+dir=data/local/local_lm
+srcdir=data/local/nist_lm
+mkdir -p $dir
+
+# check srilm ngram
+! which ngram-count \
+  && echo "SRILM tools not installed, which are required for LM training" && exit 1;
+
+# extract 5k vocabulary from a baseline language model
+srclm=$srcdir/lm_tgpr_5k.arpa.gz
+if [ -f $srclm ]; then
+  echo "Getting vocabulary from a baseline language model";
+  gunzip -c $srclm | awk 'BEGIN{unig=0}{
+    if(unig==0){
+      if($1=="\\1-grams:"){unig=1}}
+    else {
+      if ($1 != "") {
+        if ($1=="\\2-grams:" || $1=="\\end\\") {exit}
+        else {print $2}}
+    }}' > $dir/vocab_5k.txt
+else
```

Comment:
For this, srilm's ngram -write-vocab could be used -- it would perhaps make the script easier to read. You use srilm anyways... Just for your consideration